### PR TITLE
put calendars in each substitution and subreq page

### PR DESCRIPTION
**Pull request by bvt2nc**
lots of conflicts probably, please play around with things before hand

Notes:
-Calendar will only show routes with an end date... need to adapt it if there is no end date and is interval based instead
-Clean up code
-Calendar will not be created for existing users, only new users signed up after my commits will have a calendar
-Need to find solution to data calendar usage exceeded (when a large number of requests are made in a short time frame, google calendar locks the calendar into read-only mode temporarily)

### CODE REVIEW

**Comment by rcdizon** [comment]
@bvt2nc Can you take a second look at `app/app/settings.py`? Besides the fact that there's a merge conflict, there's a lot of edits that removed references to `local_settings`.

Let me know if you need a second pair of eyes, this might take a while to cross reference.

**Comment by bvt2nc** [comment]
My bad, settings.py should not have been changed. Thought I unstaged that

File: app/homepage/views.py [hunk 1]
```diff
@@ -84,10 +97,43 @@ def post(self, request, *args, **kwargs):
...
+    home_dir = os.path.expanduser('~')
+    credential_dir = os.path.join(home_dir, '.credentials')
+    if not os.path.exists(credential_dir):
+        os.makedirs(credential_dir)
+    credential_path = os.path.join(credential_dir, 'calendar-python-quickstart.json')
+
+    store = Storage(credential_path)
+    credentials = store.get()
+    if not credentials or credentials.invalid:
+        flow = client.flow_from_clientsecrets(CLIENT_SECRET_FILE, SCOPES)
```
Comment (rcdizon) [inline app/homepage/views.py]: I'm getting a `('Error opening file', 'client_secret.json', 'No such file or directory', 2)` error from this -- it seems that there's a lot of this new `client_secret_file.json` stuff going on here, any idea what might be going on?

File: app/homepage/views.py [hunk 2]
```diff
@@ -84,10 +97,43 @@ def post(self, request, *args, **kwargs):
...
+        flow.user_agent = APPLICATION_NAME
```
Comment (rcdizon) [inline app/homepage/views.py]: May have found a fix for the previous issue by assigning an absolute path, but now I'm getting an error `name 'APPLICATION_NAME' is not defined` -- what is this in reference to?

**Comment by bvt2nc** [comment]
Just made a push with a small comment to help enable it to work. I also swapped the google service account associated with it so that you guys can get the credentials to actually access the calendars when prompted. New email for it is uva.slp.meals@gmail.com. Get password from postem (the group password for pegasus meals). You should be asked to authenticate yourself with the uva.slp.meals gmail account. Afterwards, it shouldn't prompt you again unless you manually clear the credentials. If a calendar says that you don't have permissions to view the events, uncomment the line to clear credentials and then authenticate yourself again. I'll look more into not needing this work around in the future. Same notes from above apply

**Comment by rcdizon** [comment]
Will take a look at it right now, planning on just reverting the settings file

File: app/homepage/views.py [hunk 3]
```diff
@@ -34,6 +34,19 @@
 from routes.models import Route
 from app.settings import LOGO_URL
 
+import os
```
Comment (rcdizon) [inline app/homepage/views.py]: My only concern here is that there might be some weird circular dependencies somewhere here, but I'll verify this right now.

File: app/users/views.py [hunk 4]
```diff
@@ -621,6 +676,92 @@ class SubstitutionCreateView(generic.CreateView):
...
+                events = service.events().list(calendarId=calendarId, pageToken=page_token).execute()
+                for event in events['items']:
+                    service.events().delete(calendarId=calendarId, eventId=event['id']).execute()
+                page_token = events.get('nextPageToken')
+                if not page_token:
+                    break
+
+        routes = VolunteerRoute.objects.all()
+        for route in routes:
+            rrule = route.recurrences.rrules[0]
```
Comment (rcdizon) [inline app/users/views.py]: Getting an error here for `list index out of range` when I try to add a new substitute. Ideas?

**Comment by rcdizon** [comment]
Not sure why it's still showing merge conflicts -- I literally just fixed them in that last commit.

**Comment by y4ahmed** [comment]
@bvt2nc This works on Django 1.10. However on Django 1.11.6, the RadioObject used in calendar forms throws a renderer error. Will need to refactored later. Besides that, it works well and had no issues.

**Comment by ces3uf** [comment]
I'm having issues signing into the account, it says I have JavaScript disabled in my browser when I most definitely don't. I can't really verify if anything is working because of that. I also can't run the tests. If someone else can verify like Yusuf, then that's fine, but I can't do it on my end, unless someone has a workaround for this JavaScript weirdness.

**Comment by rcdizon** [comment]
@bvt2nc you can pull this in, but lets make sure to get fixes in ASAP.

**make sure to do a merge-squash so we don't bloat the commits**

tests are all working for me, so this looks good to go for now, we can revert later if we must but just make sure to close your related issues

**Comment by bvt2nc** [comment]
@y4ahmed Ah right, I forgot that they had deprecated the easy way to do horizontal radio buttons. Currently, that page isn't linked to anything so that function won't be in use for a while as that was just a sandbox page. 
@ces3uf I don't know why it would be having a JS issue as I am not using any JS unless embedding the google calendar calls some JS. I'll try to see if I can reproduce it so I can diagnose it better. 
@rcdizon Besides the above, I'll make a script by Friday to run through the existing users and create a google calendar for them since only recently created accounts have google calendars (since they are made up creation). 

If there's anything else I'm missing, let me know and I'll fix them as soon as I can.

**Pull request merged**